### ISSUE TITLE
OAuthProviderIconButtonWidget shapes

### ISSUE THREAD

**Issue by bagintz** [FirebaseExtended/flutterfire#8286]
would the team be open to a PR with options to use different shaped buttons (circles for instance)?

**Comment by darshankawar** [comment]
@bagintz 
Can you please elaborate on your proposal ?
Are you referring to `flutterfire_ui` plugin and potential changes in it ?

**Comment by bagintz** [comment]
@darshankawar Yes, the flutterfire_ui package.  I thought I had used a FirebaseUI package for a different framework (maybe Android) previously that gave you some flexibility over the login button shape.  Right now it is the squircle, I thought it would be nice to have an option for square and circle also.  I realize I can do this myself with a custom OAuthController and my own design, but using the GoogleSignInIconButton is very handy.  

The PR would introduce a `shape` option to the OAuthProviderIconButton which would be (?) based on a new enum of IconButtonShape.squircle, IconButtonShape.circle or IconButtonShape.square.  These would trigger a switch statement where it would return the correct shape (i.e. currently ClipRRect for IconButtonShape.squircle).

Alternately, it could take a custom container too that just wrapped the SvgPicture.asset.

??

**Comment by darshankawar** [comment]
Thanks for the details. I'll keep it open and label it as a proposal for the team to look into.

**Comment by bagintz** [comment]
@lesnitsky Yes, I would like to contribute, I don't think I can do it this week, but I will check it out and get something submitted in the next couple weeks.
